docs/ldml/tr35-keyboards.md

```diff
@@ -874,327 +1002,236 @@ The `display` element describes how a character, that has come from a `keyMap/ma
+> to choose a different keyboard file, or use some other fallback behavior when using a
+> hardware keyboard.
+>
+> When using an on-screen keyboard, if there is not a `<layerMaps form="touch">`
+> element, the `<layerMaps form="hardware">` element can be used for on-screen use.
 
-```
-keyId = ISOKey | specialKey
-ISOKey = [A-Z][0-9][0-9]
-specialKey = [a-z][a-zA-Z0-9]{2,7}
-```
+### 5.13.1 <a name="Element_layerMap" href="#Element_layerMap">Element: layerMap</a>
 
-ISOKey denotes a key having an [ISO Position](#Definitions). SpecialKey is used to identify functional keys occurring on a virtual keyboard layout.
+A `layerMap` element describes the configuration of keys on a particular layer of a keyboard. It contains one or more `row` elements to describe which keys exist in each row.
 
 **Syntax**
 
 ```xml
-<row keys="{keyId}" />
+<layerMap id="layerId" modifier="{Set of Modifier Combinations}">
+    ...
+</layerMap>
 ```
 
 > <small>
 >
-> Parents: [layer](#Element_layer)
-> Children: _none_
-> Occurence: required, multiple
+> Parents: [keyboard](#Element_keyboard)
+> Children: [row](#Element_row)
+> Occurrence: optional, multiple
 >
 > </small>
 
-_Attribute:_ `keys` (required)
+_Attribute_ `id` (required for `touch`)
 
-> This is a string that lists the `keyId` for each of the keys in a row. Key ranges may be contracted to firstkey-lastkey but only for `ISOKey` type `keyId`s. The interpolation between the first and last keys names is entirely numeric. Thus `D00-D03` is equivalent to `D00 D01 D02 D03`. It is an error if the first and last keys do not have the same alphabetic prefix or the last key numeric component is less than or equal to the first key numeric component.
+> The `id` attribute identifies the layer for touch layouts.  This identifier specifies the layout as the target for layer switching, as specified by the `switch=` attribute on the [`<key>`](#Element_key) element.
+> Touch layouts must have one `layerMap` with `id="base"` to serve as the base layer.
 
-`specialKey` type `keyId`s may take any value within their syntactic constraint. But the following `specialKey`s are reserved to allow applications to identify them and give them special handling:
+_Attribute:_ `modifier` (required for `hardware`)
 
-* `"bksp"`, `"enter"`, `"space"`, `"tab"`, "`esc"`, `"sym"`, `"num"`
-* all the reserved modifier names
-* specialKeys starting with the letter "x" for future reserved names.
+> This has two roles. It acts as an identifier for the `layer` element for hardware keyboards (in the absence of the id= element) and also provides the linkage from the hardware modifiers into the correct `layer`. To indicate that no modifiers apply the reserved name of "none" can be used. For the purposes of fallback vkey mapping, the following modifier components are reserved: "shift", "ctrl", "alt", "caps", "cmd", "opt" along with the "L" and "R" optional single suffixes for the first 3 in that list.
```

Comment:
> To indicate that no modifiers apply the reserved name...

A comma would be appreciated between "modifiers" and "apply".  That sentence was a bit tricky to parse on first read.

Reply:
- [ ] Comma